KubeArmor/core/kubeUpdate.go

```diff
@@ -360,19 +362,17 @@ func (dm *KubeArmorDaemon) UpdateEndPointWithPod(action string, pod tp.K8sPod) {
 			}
 		}
 
-		dm.EndPointsLock.Unlock()
-
 	} else if action == "MODIFIED" {
 		newEndPoint := tp.EndPoint{}
 		endpoints := []tp.EndPoint{}
 
-		dm.EndPointsLock.Lock()
+		dm.EndPointsLock.RLock()
 		for _, endPoint := range dm.EndPoints {
 			if pod.Metadata["namespaceName"] == endPoint.NamespaceName && pod.Metadata["podName"] == endPoint.EndPointName {
 				endpoints = append(endpoints, endPoint)
 			}
 		}
-		dm.EndPointsLock.Unlock()
+		dm.EndPointsLock.RUnlock()
 		if len(endpoints) == 0 {
```

Comment:
Is the only goal for the above for loop is just to check if it's not 0? it should be fine to break out of loop once we add one endpoint